test/test_other.py

```diff
@@ -8656,21 +8675,21 @@ def test_metadce_minimal_pthreads(self):
     self.run_metadce_test('minimal_main.c', ['-Oz', '-pthread', '-sPROXY_TO_PTHREAD'])
 
   @parameterized({
-    'noexcept': (['-O2'],                    [], ['waka']), # noqa
+    'noexcept': (['-Oz'],                    [], ['waka']), # noqa
     # exceptions increases code size significantly
-    'except':   (['-O2', '-fexceptions'],    [], ['waka']), # noqa
+    'except':   (['-Oz', '-fexceptions'],    [], ['waka']), # noqa
     # exceptions does not pull in demangling by default, which increases code size
-    'mangle':   (['-O2', '-fexceptions',
+    'mangle':   (['-Oz', '-fexceptions',
                   '-sDEMANGLE_SUPPORT', '-Wno-deprecated'], [], ['waka']), # noqa
     # Wasm EH's code size increase is smaller than that of Emscripten EH
-    'except_wasm':   (['-O2', '-fwasm-exceptions'], [], ['waka']),
-    'except_wasm_exnref':   (['-O2', '-fwasm-exceptions', '-sWASM_EXNREF'], [], ['waka']),
+    'except_wasm':   (['-Oz', '-fwasm-exceptions'], [], ['waka']),
+    'except_wasm_exnref':   (['-Oz', '-fwasm-exceptions', '-sWASM_EXNREF'], [], ['waka']),
     # eval_ctors 1 can partially optimize, but runs into getenv() for locale
     # code. mode 2 ignores those and fully optimizes out the ctors
-    'ctors1':    (['-O2', '-sEVAL_CTORS'],   [], ['waka']),
-    'ctors2':    (['-O2', '-sEVAL_CTORS=2'], [], ['waka']),
-    'wasmfs':    (['-O2', '-sWASMFS'],       [], ['waka']),
-    'lto':       (['-O2', '-flto'],       [], ['waka']),
+    'ctors1':    (['-Oz', '-sEVAL_CTORS'],   [], ['waka']),
+    'ctors2':    (['-Oz', '-sEVAL_CTORS=2'], [], ['waka']),
+    'wasmfs':    (['-Oz', '-sWASMFS'],       [], ['waka']),
+    'lto':       (['-Oz', '-flto'],       [], ['waka']),
```

Comment:
I think this might be excessive. While the tests are called "metadce" they are really code size tests. Code size of metadce is important as higher opt levels do it, but there is also value to testing medium levels like `-O2`. Also, `-O2` is likely more common than `-Oz`.

Reply:
OK, I'll maybe only do this for the `-flto` one then?

Also, we should rename this infra to reflect that these are not really just metadce tests.